tools/integration-test/src/framework/nary/mod.rs

```diff
@@ -0,0 +1,13 @@
+/*!
+   Experimental work to define test cases with N-ary chains.
```

Comment:
I added the N-ary test chains implementation in this PR as well, but hide is behind an `"experimental"` flag so that we can still merge this even though it is not too ready for review yet due to documentation not yet being written.

Alternatively I can move the N-ary modules into a separate draft PR so that they can be reviewed separately.